### PR TITLE
[cryptofuzz] Add Botan

### PR DESCRIPTION
**Pull request by guidovranken**
Also disabling libsodium because it currently produces crashing builds.

### CODE REVIEW

**Comment by guidovranken** [comment]
It is possible that Travis will fail due to timeout.

**Comment by guidovranken** [comment]
@randombit Would you like to be added to the CC's? This means that you'll receive bug reports of bugs found by Cryptofuzz/OSS-Fuzz. Note that there may be a lot of noise due to bugs in other libraries (particularly OpenSSL which is under heavy development and frequently breaks something); alternatively I can manually forward bug reports relevant to Botan to you.

File: projects/cryptofuzz/build.sh [hunk 1]
```diff
@@ -127,22 +146,22 @@ then
...
-
-    export CXXFLAGS="$CXXFLAGS -DCRYPTOFUZZ_LIBSODIUM"
-    export LIBSODIUM_A_PATH="$SRC/libsodium/src/libsodium/.libs/libsodium.a"
-    export LIBSODIUM_INCLUDE_PATH="$SRC/libsodium/src/libsodium/include"
-
-    # Compile Cryptofuzz libsodium (with assembly) module
-    cd $SRC/cryptofuzz/modules/libsodium
-    make -B
-fi
+#if [[ $CFLAGS != *sanitize=memory* ]]
```
Comment (oliverchang) [inline projects/cryptofuzz/build.sh]: nit: add a comment explaining why this is commented.

Comment (guidovranken) [reply projects/cryptofuzz/build.sh]: Done

**Comment by guidovranken** [comment]
It would be nice if you would change the timeouts of Travis and/or use a better CPU. Even on my machine with 12 cores Cryptofuzz with all modules takes a long time to compile.

**Comment by inferno-chromium** [comment]
> It would be nice if you would change the timeouts of Travis and/or use a better CPU. Even on my machine with 12 cores Cryptofuzz with all modules takes a long time to compile.

Our travis timeout is already increased to 3 hrs now. Is it a problem that nothing is printed to log for some time, can you try this travis_wait trick ?
https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

**Comment by randombit** [comment]
@guidovranken CC would be lovely thank you - jack.lloyd@gmail.com. Even bugs that don't affect us will give me ideas for additional tests that can prevent future regressions.

I think the problem with CI is just the Botan build is taking > 10 minutes and the output  of `make` is being redirected to /dev/null so Travis thinks it has stalled.

Re build times, maybe ccache would help?

**Comment by inferno-chromium** [comment]
we have got the travis timeout for no build output from 10 min to 30 min, so hopefully travis succeeds.

**Comment by guidovranken** [comment]
```
[*] Loading extra dictionary from 'cryptofuzz-openssl-noasm.dict' (level 0)...
[-] PROGRAM ABORT : Keyword too big in line 314 (137 B, limit is 128 B)

         Location : load_extras_file(), afl-fuzz.c:1656
```

Hahaha, so 1940s AFL.

Fixed and rebuilding..

**Comment by guidovranken** [comment]
> we have got the travis timeout for no build output from 10 min to 30 min, so hopefully travis succeeds.

Thank you. One more thing that would be nice to have fixed (assuming it's still the same as before) is failing on many log lines. This is why I had to resort to directing output to /dev/null at some point. This doesn't have priority though.

**Comment by guidovranken** [comment]
All builds succeed, we're good to go.

**Comment by jonathanmetzman** [comment]
> This is why I had to resort to directing output to /dev/null at some point. This doesn't have priority though.

Heh. I think I suggested this.

**Comment by guidovranken** [comment]
> > This is why I had to resort to directing output to /dev/null at some point. This doesn't have priority though.
> 
> Heh. I think I suggested this.

not denying that

**Pull request merged**